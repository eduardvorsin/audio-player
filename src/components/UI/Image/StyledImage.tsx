import styled from "styled-components";

interface StyledImageProps {
	className?: string
}

export const StyledImage = styled.img<StyledImageProps>`
	object-fit: cover;
	height: auto;
`;