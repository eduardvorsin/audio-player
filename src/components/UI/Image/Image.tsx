import React from 'react';
import { StyledImage } from './StyledImage';


interface ImageProps extends Partial<React.ImgHTMLAttributes<HTMLImageElement>> {
	src: string,
	alt: string,
	width?: number,
	height?: number,
	className?: string,
};

export const Image: React.FC<ImageProps> = ({
	src,
	alt,
	width = 50,
	height = 50,
	className = '',
	...props
}) => {
	return (
		<StyledImage
			src={src}
			width={width}
			height={height}
			alt={alt}
			className={className}
			{...props}
		/>
	);
};