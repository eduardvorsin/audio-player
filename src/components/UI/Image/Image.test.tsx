import { render, screen } from "@testing-library/react";
import React from "react";
import { Image } from "./Image";

describe('Image component tests', () => {

  test('renders correctly', () => {
    render(
      <Image
        src='https://via.placeholder.com/350x150'
        alt='placeholder image'
      />
    );
    expect(screen.getByAltText<HTMLImageElement>(/placeholder image/i)).toBeInTheDocument();
  });

  test('the image is rendered with a specific width', () => {
    render(
      <Image
        src='https://via.placeholder.com/200x200'
        width={200}
        alt='placeholder image'
      />
    );

    expect(screen.getByAltText<HTMLImageElement>(/placeholder image/i)).toHaveStyle('width:200');
  });

  test('the image is rendered with a specific height', () => {
    render(
      <Image
        src='https://via.placeholder.com/150x150'
        height={150}
        alt='placeholder image'
      />
    );

    expect(screen.getByAltText<HTMLImageElement>(/placeholder image/i)).toHaveStyle('width:150');
  });

  test('the image is rendered with a specific alt text', () => {
    render(
      <Image
        src='https://via.placeholder.com/200x200'
        width={200}
        height={200}
        alt='placeholder image'
      />
    );

    expect(screen.getByAltText<HTMLImageElement>(/placeholder image/i)).toBeInTheDocument();
  });

  test('className is assigned correctly', () => {
    render(
      <Image
        src='https://via.placeholder.com/200x200'
        className='test'
        alt='placeholder image'
      />
    );

    expect(screen.getByAltText<HTMLImageElement>(/placeholder image/i)).toHaveClass('test');
  });

  test('snapshot without optional props', () => {
    render(
      <Image
        src='https://via.placeholder.com/100x100'
        alt='placeholder image'
      />
    );

    expect(screen.getByAltText<HTMLImageElement>('placeholder image')).toMatchSnapshot();
  });

  test('snapshot with all the passed props', () => {
    render(
      <Image
        src='a.png'
        width={20}
        height={20}
        className='test'
        alt='new placeholder image'
      />
    );

    expect(screen.getByAltText<HTMLImageElement>(/new placeholder image/i)).toMatchSnapshot();
  });
});
