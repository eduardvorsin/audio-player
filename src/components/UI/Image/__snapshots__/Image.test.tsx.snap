// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component tests snapshot with all the passed props 1`] = `
<img
  alt="new placeholder image"
  class="sc-bcXHqe fZAhUH test"
  height="20"
  src="a.png"
  width="20"
/>
`;

exports[`Image component tests snapshot without optional props 1`] = `
<img
  alt="placeholder image"
  class="sc-bcXHqe fZAhUH"
  height="50"
  src="https://via.placeholder.com/100x100"
  width="50"
/>
`;
